xla/service/gpu/runtime/gpublas_lt_matmul.h

```diff
@@ -35,9 +35,21 @@ void PopulateCublasLtMatmulAttrEncoding(
 // Registers XLA Gpu runtime cuBLASLt custom calls.
 void RegisterMatmulCustomCalls(runtime::DirectCustomCallRegistry& registry);
 
+struct MatmulPlanVec final {
+    using Result = absl::StatusOr<se::gpu::BlasLt::MatmulPlan *>;
+    using CreateFunc = absl::AnyInvocable<
+                        absl::StatusOr<se::gpu::BlasLt::MatmulPlanPtr>()>;
+
+    explicit MatmulPlanVec(size_t size) : plans_(size), mu_{new absl::Mutex} {}
+    
+    Result GetOrCreate(int device_id, CreateFunc create_plan);
+private: 
+  std::vector<se::gpu::BlasLt::MatmulPlanPtr> plans_;  
+  // unique_ptr is usually not needed. But maybe in your case?  
```

Comment:
Nit: This comment is not needed, I just added it as a question for you in the comment.